mobify/progressive-web-scaffold
# Add image optimization

Add a task to optimize images.

## Changes
- Added `imagemin-power-cli` as a dev dependency
- Cleaned up some inconsistencies in package.json, including
    - Re-arrange some non-alphabetically ordered dependencies
    - Change `clean-svgs` to `clean:svgs` to match the new `clean:images` task
- Optimized images. Compression savings are as follows:

```
✔ Minifying image "./app/static/img/categories/books.png" (1 of 31) - saved 1.32 kB - 27.7%...
✔ Minifying image "./app/static/img/categories/books@2x.png" (2 of 31) - saved 1.74 kB - 33.8%...
✔ Minifying image "./app/static/img/categories/charms.png" (3 of 31) - saved 1.54 kB - 31.7%...
✔ Minifying image "./app/static/img/cart/empty-cart.png" (4 of 31) - saved 3.33 kB - 39.4%...
✔ Minifying image "./app/static/img/categories/ingredients.png" (5 of 31) - saved 1.3 kB - 28.7%...
✔ Minifying image "./app/static/img/categories/charms@2x.png" (6 of 31) - saved 3.02 kB - 38.6%...
✔ Minifying image "./app/static/img/categories/new-arrivals.png" (7 of 31) - saved 1.99 kB - 29.3%...
✔ Minifying image "./app/static/img/cart/wishlist@2x.png" (8 of 31) - saved 5.36 kB - 46.4%...
✔ Minifying image "./app/static/img/categories/ingredients@2x.png" (9 of 31) - saved 2.2 kB - 34.4%...
✔ Minifying image "./app/static/img/categories/potions.png" (10 of 31) - saved 1.1 kB - 28.9%...
✔ Minifying image "./app/static/img/categories/supplies.png" (11 of 31) - saved 1.38 kB - 28.6%...
✔ Minifying image "./app/static/img/categories/potions@2x.png" (12 of 31) - saved 1.9 kB - 35.5%...
✔ Minifying image "./app/static/img/categories/new-arrivals@2x.png" (13 of 31) - saved 3.67 kB - 38.2%...
✔ Minifying image "./app/static/img/categories/supplies@2x.png" (14 of 31) - saved 2.51 kB - 35.4%...
✔ Minifying image "./app/static/img/checkout/cc-american-express@3x.png" (15 of 31) - saved 3.08 kB - 39.6%...
✔ Minifying image "./app/static/img/checkout/cc-discover@3x.png" (16 of 31) - saved 1.84 kB - 32.3%...
✔ Minifying image "./app/static/img/cart/empty-cart@2x.png" (17 of 31) - saved 7.72 kB - 41.5%...
✔ Minifying image "./app/static/img/checkout/account-created.png" (18 of 31) - saved 4.57 kB - 47.3%...
✔ Minifying image "./app/static/img/checkout/cc-visa@3x.png" (19 of 31) - saved 1.89 kB - 34.8%...
✔ Minifying image "./app/static/img/checkout/hint-amex@3x.png" (20 of 31) - saved 2.34 kB - 49.5%...
✔ Minifying image "./app/static/img/checkout/cc-unionpay@3x.png" (21 of 31) - saved 2.28 kB - 26.1%...
✔ Minifying image "./app/static/img/checkout/hint-visa-mc@3x.png" (22 of 31) - saved 1.27 kB - 58%...
✔ Minifying image "./app/static/img/checkout/confirmed.png" (23 of 31) - saved 3.67 kB - 43.1%...
✔ Minifying image "./app/static/img/checkout/cc-mastercard@3x.png" (24 of 31) - saved 2.29 kB - 25.9%...
✔ Minifying image "./app/static/img/checkout/verisign-mcafee-secure.png" (25 of 31) - saved 3.25 kB - 16%...
✔ Minifying image "./app/static/img/global/logo.png" (26 of 31) - saved 210 B - 4.2%...
✔ Minifying image "./app/static/img/global/no-results.png" (27 of 31) - saved 8.24 kB - 44.6%...
✔ Minifying image "./app/static/img/global/app-icon.png" (28 of 31) - saved 17.8 kB - 50.6%...
✔ Minifying image "./app/static/img/homepage_carousel/0.png" (29 of 31) - saved 271 kB - 29.6%...
✔ Minifying image "./app/static/img/homepage_carousel/2.png" (30 of 31) - saved 185 kB - 22.9%...
✔ Minifying image "./app/static/img/homepage_carousel/1.png" (31 of 31) - saved 209 kB - 27.2%...
```

```
ℹ Successfully compressed images: 31. Unsuccessfully compressed images: 0. Total images: 31. Total images size: 2.73 MB. Total saved size: 757 kB - 27.675588830646102%. 
```

## How to test-drive this PR
- Run `npm run clean:images`
- Run `npm run dev` and [preview Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Confirm that images work as intended.
